commands/transfer_funds.go

```diff
@@ -14,35 +20,77 @@ func checkTransfer(cmd *commandspb.Transfer) Errors {
 	errs := NewErrors()
 
 	if cmd == nil {
-		return errs.FinalAddForProperty("transfer_funds", ErrIsRequired)
+		return errs.FinalAddForProperty("transfer", ErrIsRequired)
 	}
 
 	if len(cmd.Amount) <= 0 {
-		errs.AddForProperty("transfer_funds.amount", ErrIsRequired)
+		errs.AddForProperty("transfer.amount", ErrIsRequired)
 	} else {
 		if amount, ok := big.NewInt(0).SetString(cmd.Amount, 10); !ok {
-			errs.AddForProperty("transfer_funds.amount", ErrNotAValidInteger)
+			errs.AddForProperty("transfer.amount", ErrNotAValidInteger)
 		} else {
 			if amount.Cmp(big.NewInt(0)) == 0 {
-				errs.AddForProperty("transfer_funds.amount", ErrIsRequired)
+				errs.AddForProperty("transfer.amount", ErrIsRequired)
 			}
 			if amount.Cmp(big.NewInt(0)) == -1 {
-				errs.AddForProperty("transfer_funds.amount", ErrMustBePositive)
+				errs.AddForProperty("transfer.amount", ErrMustBePositive)
 			}
 		}
 	}
 
 	if len(cmd.To) <= 0 {
-		errs.AddForProperty("transfer_funds.to", ErrIsRequired)
+		errs.AddForProperty("transfer.to", ErrIsRequired)
+	}
+
+	if cmd.ToAccountType == vega.AccountType_ACCOUNT_TYPE_UNSPECIFIED {
+		errs.AddForProperty("transfer.to_account_type", ErrIsNotValid)
+	} else if _, ok := vega.AccountType_name[int32(cmd.ToAccountType)]; !ok {
+		errs.AddForProperty("transfer.to_account_type", ErrIsNotValid)
+	}
+
+	if cmd.FromAccountType == vega.AccountType_ACCOUNT_TYPE_UNSPECIFIED {
```

Comment:
not sure if it this is the place to verify but we also need to make sure we do not allow to transfer *from* one of the reward accounts.

Reply:
This is checked on the core.

Reply:
it's more a note to self to remember to check this :)

Reply:
ah fair enough